src/characteristic/abstractSplitTemperatureThresholdCharacteristic.ts

```diff
@@ -22,58 +27,62 @@ export default class AbstractSplithresholdCharacteristic extends AbstractCharact
   typeof Characteristic.CoolingThresholdTemperature
 > {
   mode: Mode
-  localPlatform: HomebridgeLgThinqPlatform
-  localService: Service
 
   constructor(
     platform: HomebridgeLgThinqPlatform,
     service: Service,
-    deviceId: string,
+    device: LgAirConditionerPlatformAccessory,
     mode: Mode,
   ) {
     super(
       platform,
       service,
-      deviceId,
+      device,
       mode === 'cool'
         ? platform.Characteristic.CoolingThresholdTemperature
         : platform.Characteristic.HeatingThresholdTemperature,
       'Set',
       'airState.tempState.target',
     )
+
     this.mode = mode
     service
       .getCharacteristic(this.characteristic)
       // min/max as defined in product manual
-      .setProps({ minValue: 16, maxValue: 30, minStep: 0.5 })
-    // Usually these would be private, but this is a special characteristic
-    // that needs these
-    this.localPlatform = platform
-    this.localService = service
+      .setProps({
+        minValue: 16,
+        maxValue: 30,
+        minStep: 0.5,
+      })
   }
 
-  // Override default handleUpdatedSnapshot() to ignore based on mode
-  handleUpdatedSnapshot(snapshot: GetDeviceResponse['result']['snapshot']) {
-    const targetState = this.localService.getCharacteristic(
-      this.localPlatform.Characteristic.TargetHeaterCoolerState,
-    ).value
-    const requiredState =
-      this.mode === 'cool'
-        ? this.localPlatform.Characteristic.TargetHeaterCoolerState.COOL
-        : this.localPlatform.Characteristic.TargetHeaterCoolerState.HEAT
-    if (targetState !== requiredState) {
-      this.logDebug(
-        `Target state is not "${this.mode}", ignoring snapshot update`,
-      )
+  handleSet?(value: CharacteristicValue, callback: CharacteristicSetCallback) {
+    if (this.device.lockTemperature) {
+      // updating from cache when we're locked puts things back to where they were, essentially preventing edits
+      this.device.updateCharacteristics(true)
```

Comment:
Okay, but then the user can change the temperature to something the AC won't accept, and the temperature will just revert back next refresh. Which could be 10 minutes from now. I didn't think that was great either...

Reply:
I sort of think of it as one action -> one failure (ignored input) being better than (potentially) one action -> stale UI due to cache -> in-sync UI moments later. It's unfortunate that HomeKit doesn't support this set of modes appropriately but I think it could be more confusing to flash the user with a stale UI for this one case

Reply:
Got it. You should try disabling the "lockTemperature" flag and run the plugin to see what happens without it--it's not great. Basically you still get what you could be construed as a cached update: the temperature stays where the user set it, then snaps back to what the AC is actually at next refresh, potentially 10 minutes later. So my thinking here is that the user thinks they set the temperature to something preferable, then later realize the update didn't work. This would happen 100% of the time, whereas the cache being stale would be only when the connection between Thinq and the plugin is down, which I assumed was less than that. So TL;DR: this approach shows the correct state of the AC more often than any alternative. 

I agree the real fix would be on the HomeKit side, but alas, can't find any fix there. Know anyone at Apple to ask what they would do in this case? Maybe there's something we don't know here...